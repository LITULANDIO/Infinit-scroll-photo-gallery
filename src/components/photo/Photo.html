    
    <div class="cursor-pointer" @click="onClicked">
        <img :src="photo.thumbnailUrl" :alt="photo.id" />
    </div>
