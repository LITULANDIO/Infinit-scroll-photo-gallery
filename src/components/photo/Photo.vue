<template src="./Photo.html"></template>

<script>
export default {
    name: 'Photo',
    props:{
        photo:{
            type: Object,
            required: true
        }
    },
    setup(props, {emit}){
        const onClicked = () =>{
            emit('onClicked', props.photo)
        }

    
        return{ 
            onClicked,
        }
    }
}
</script>

<style lang="scss" src="./Photo.scss" scoped></style>