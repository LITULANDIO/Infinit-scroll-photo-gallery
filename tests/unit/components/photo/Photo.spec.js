import { shallowMount } from '@vue/test-utils';
import Photo from '@/components/photo/Photo.vue';
import photo from '../../mocks/photo.mock';

describe('Photo component', () =>{

    test('must match snapshot', () =>{
        const wrapper = shallowMount(Photo, {
            props:{
                photo
            }
        })
        expect(wrapper.html()).toMatchSnapshot()
    })

    test('the image id exists', () =>{
        const wrapper = shallowMount(Photo, {
            props:{
                photo
            }
        })
        const image = wrapper.find('img');
        expect(image.attributes('src')).toBe(undefined)
        expect(image.attributes('alt')).toBe('1')

        const imgList = wrapper.findAll('img')

        // check images
        expect(imgList.length).toBe(4);

        const [ img1, img2, img3, img4 ] = imgList;

        // check event click
        img1.trigger('click');
        
        // check emmited
        expect(wrapper.emitted('onClicked'))



    })

})
