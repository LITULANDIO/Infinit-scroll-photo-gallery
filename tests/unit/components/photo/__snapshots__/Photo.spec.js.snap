// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Photo component must match snapshot 1`] = `<div class="cursor-pointer"><img alt="1"></div>`;
